src/middlewared/middlewared/plugins/jail.py

```diff
@@ -117,6 +580,45 @@ def query(self, filters=None, options=None):
 
         return filter_list(jails, filters, options)
 
+    @accepts(
+        Bool('remote', default=False),
+    )
+    def releases_choices(self, remote):
+        """
+        List installed or available releases which can be downloaded.
+        """
+        if remote:
+            with contextlib.suppress(KeyError):
+                return self.middleware.call_sync('cache.get', 'iocage_remote_releases')
+
+        iocage = ioc.IOCage(skip_jails=True)
+        choices = {k: k for k in iocage.fetch(list=True, remote=remote, http=True)}
+
+        if remote:
+            self.middleware.call_sync('cache.put', 'iocage_remote_releases', choices, 86400)
+
+        return choices
+
+    @accepts(
+        Dict(
+            'branches_choices',
+            Str('account', default='freenas'),
+            Str('repository', default='iocage-ix-plugins'),
+        )
+    )
+    def branches_choices(self, options):
+        """
+        List branches for a github `account`'s `repository`.
```

Comment:
Not a fan of relying on github for that. Is there anything we can do about that?